### PR TITLE
`UninterpretedConstOracle` refactor

### PR DESCRIPTION
**Pull request by Kukovec**
- [x] Tests added for any new code
- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [ ] Documentation added for any new functionality
- [ ] [Entries added to `./unreleased/`][changelog format] for any new functionality

Following the pattern laid out in #2709 for `IntOracle`, this PR introduces the same alterations to `UninterpretedConstOracle`.

[changelog format]: https://github.com/informalsystems/apalache/blob/main/CONTRIBUTING.md#how-to-record-a-change

### CODE REVIEW

**Comment by Kukovec** [comment]
@konnov can you take a look at this soon-ish? I'd like to close this (and #2731 after)

File: tla-bmcmt/src/test/scala/at/forsyte/apalache/tla/bmcmt/stratifiedRules/aux/oracles/TestUCOracle.scala [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+import org.junit.runner.RunWith
+import org.scalacheck.Prop.forAll
+import org.scalacheck.{Gen, Prop}
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatestplus.junit.JUnitRunner
+import org.scalatestplus.scalacheck.Checkers
+
+@RunWith(classOf[JUnitRunner])
+class TestUCOracle extends AnyFunSuite with BeforeAndAfterEach with Checkers {
```
Comment (thpani) [inline tla-bmcmt/src/test/scala/at/forsyte/apalache/tla/bmcmt/stratifiedRules/aux/oracles/TestUCOracle.scala:21]: Shortening the class name makes this hard to find (if I'm looking for the tests for `UninterpretedConstOracle`).

```suggestion
class TestUninterpretedConstOracle extends AnyFunSuite with BeforeAndAfterEach with Checkers {
```

**Review by thpani — APPROVED** [review]
LGTM

File: tla-bmcmt/src/test/scala/at/forsyte/apalache/tla/bmcmt/stratifiedRules/aux/oracles/TestUCOracle.scala (hunk 1 above)
Comment (Kukovec) [reply tla-bmcmt/src/test/scala/at/forsyte/apalache/tla/bmcmt/stratifiedRules/aux/oracles/TestUCOracle.scala:21]: Good point. I'll have to add this manually though to also rename the file.

**Pull request merged**